execution/record.go

```diff
@@ -80,10 +86,11 @@ func NewRecordFromSlice(fields []octosql.VariableName, data []octosql.Value, opt
 }
 
 func NewRecordFromRecord(record *Record, opts ...RecordOption) *Record {
+	metadataCopy := *record.Metadata
 	r := &Record{
 		FieldNames: record.FieldNames,
 		Data:       record.Data,
-		Metadata:   &(*record.Metadata),
+		Metadata:   &metadataCopy,
```

Comment:
Out of curiosity - what is the reason behind dereferencing a pointer that is record.Metadata and then taking the new address? I mean I understand that we end up having the same value, but with a different address, but why?

Reply:
Because this new metadata may be modified using options like WithNoUndo. This way we won't accidentally modify metadata from other record. We never modify data or fields, so we can reuse those without issue.